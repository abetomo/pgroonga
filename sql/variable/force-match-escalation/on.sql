CREATE TABLE memos (
  id integer,
  content text
);

INSERT INTO memos VALUES (1, 'PostgreSQL is a RDBMS.');
INSERT INTO memos VALUES (2, 'Groonga is fast full text search engine.');
INSERT INTO memos VALUES (3, 'PGroonga is a PostgreSQL extension that uses Groonga.');

CREATE INDEX grnindex ON memos
 USING pgroonga (content pgroonga_text_full_text_search_ops_v2);

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;

SHOW pgroonga.force_match_escalation;
EXPLAIN (COSTS OFF)
SELECT id, content
  FROM memos
 WHERE content &@~ 'gro eng';
SELECT id, content
  FROM memos
 WHERE content &@~ 'gro eng';

SET pgroonga.force_match_escalation = on;
SHOW pgroonga.force_match_escalation;
EXPLAIN (COSTS OFF)
SELECT id, content
  FROM memos
 WHERE content &@~ 'gro eng';
SELECT id, content
  FROM memos
 WHERE content &@~ 'gro eng';

SET pgroonga.force_match_escalation = default;

DROP TABLE memos;
