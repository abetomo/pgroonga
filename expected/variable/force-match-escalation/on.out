CREATE TABLE memos (
  id integer,
  content text
);
INSERT INTO memos VALUES (1, 'PostgreSQL is a RDBMS.');
INSERT INTO memos VALUES (2, 'Groonga is fast full text search engine.');
INSERT INTO memos VALUES (3, 'PGroonga is a PostgreSQL extension that uses Groonga.');
CREATE INDEX grnindex ON memos
 USING pgroonga (content pgroonga_text_full_text_search_ops_v2);
SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
SHOW pgroonga.force_match_escalation;
 pgroonga.force_match_escalation 
---------------------------------
 off
(1 row)

EXPLAIN (COSTS OFF)
SELECT id, content
  FROM memos
 WHERE content &@~ 'gro eng';
                 QUERY PLAN                  
---------------------------------------------
 Index Scan using grnindex on memos
   Index Cond: (content &@~ 'gro eng'::text)
(2 rows)

SELECT id, content
  FROM memos
 WHERE content &@~ 'gro eng';
 id | content 
----+---------
(0 rows)

SET pgroonga.force_match_escalation = on;
SHOW pgroonga.force_match_escalation;
 pgroonga.force_match_escalation 
---------------------------------
 on
(1 row)

EXPLAIN (COSTS OFF)
SELECT id, content
  FROM memos
 WHERE content &@~ 'gro eng';
                 QUERY PLAN                  
---------------------------------------------
 Index Scan using grnindex on memos
   Index Cond: (content &@~ 'gro eng'::text)
(2 rows)

SELECT id, content
  FROM memos
 WHERE content &@~ 'gro eng';
 id |                 content                  
----+------------------------------------------
  2 | Groonga is fast full text search engine.
(1 row)

SET pgroonga.force_match_escalation = default;
DROP TABLE memos;
